samples: nrf9160: gnss: update reference position support

tokangas opened the pull request:
Small improvements for the reference position support based on PR review comments which were submitted after PR #7505 was already merged.

NordicBuilder commented:
### Integration test specification

| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-nrf-iot_positioning | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf/job/master/23076/artifact/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

tokangas commented:
Rebased and added a new commit. The new commit changes all printk() calls to printf() calls. The reason for this change is, that printk() doesn't print floating point numbers correctly when FPU is disabled, while printf() does.
